chapters/solutions.Rmd

```diff
@@ -677,15 +691,25 @@ ENIAC was the world's first general-purpose electronic computer.
 
 ### Exercise \@ref(git-cmdline-ex-diff) {-}
 
-FIXME: solution for exercise on git diff
+1. `git diff HEAD~9 bin/plotcounts.py` compares what has changed between the 
+current `bin/plotcounts.py` and the same file 9 commits ago.
+2.  The previous git command takes the state of the file at each point and then
+compares them.
+3.  `git diff HEAD bin/plotcounts.py` compares what has been changed in
+`bin/plotcounts.py` with the previous commit.
 
 ### Exercise \@ref(git-cmdline-ex-unstage) {-}
 
-FIXME: solution for exercise on git unstage
+Using `git checkout` on a staged file does not unstage it. That's 
+because the changes are in the staged area and checkout would affect
+the working directory.
 
 ### Exercise \@ref(git-cmdline-ex-blame) {-}
 
-FIXME: solution for exercise on git blame
+1. Each line of output corresponds to a line in the file, who was last
+to modify the line, when that change was made, and what the file name was
+or is called.
+2. FIXME: I don't see any ^ when I use it on a random file in the book repo.
```

Comment:
The ^ designates the boundary commit. (In other words, the lines of the file that have a ^ haven't changed since the initial commit.)
https://stackoverflow.com/questions/13105858/why-would-a-lines-sha-in-a-git-blame-view-have-a-leading-caret

If you run `git blame bin/plotcounts.py` at this point of the book (when there's only been one committed change to `plotcounts.py` since the initial commit) almost every line has a ^

Reply:
good point - I don't know how to amend the exercise given the state of the repo at this point in the book, but I think `git blame` is important. Maybe move this exercise to `git-advanced` when there will be more history to explore?

Reply:
I made all the other changes but I'll leave this for now. Should we move it? Let me know and I will.